fairscale/nn/data_parallel/fsdp_optim_utils.py

```diff
@@ -100,7 +117,7 @@ def _unflatten_optim_state(
 
     # non_tensor_state refers to entries in sd[state][param_id] that are not tensors, like "step".
     # we check that these are identical across workers and then take the first
-    non_tensor_state = [_extract_non_tensor_state(combined_state, id) for id in combined_state]
+    non_tensor_state = {id: _extract_non_tensor_state(combined_state, id) for id in combined_state}
```

Comment:
good question. I am not 100% sure. I am still trying to document this part of the code as much as I could. I think it is always 0 because it has only 1 element. See assert on line 170 below